### PR TITLE
Better support new assembly

### PR DESCRIPTION
**Pull request by thomasmaurel**
Added new checks using the core meta table and metadata database to find out if a handed over database is on a new assembly. If the assembly is the same then the handover service will carry on with the database. If the species is not found in the metadata database then we will consider it as a new species, the update_type flag will be set to new_assembly. If the assembly is different between the core database and the metadata database, set the update_flag to new assembly and gather the list of old assembly databases. Then before copying the new database, drop the old assembly databases from staging. I have also added more comments to methods and removed unnecessary brackets in if/else conditions. Fixed typo in hive.py file.

### CODE REVIEW

File: ensembl_prodinf/handover_tasks.py [hunk 1]
```diff
@@ -71,18 +73,21 @@ def handover_database(spec):
...
         raise ValueError(spec['src_uri'] + " has been handed over after the deadline. Please contact the Production team")
+    #Check if this is a new assembly
+    spec = check_new_assembly(spec,cfg.metadata_uri,db_prefix,release,db_type)
     #Get database hc group and compara_uri
     (groups,compara_uri) = hc_groups(db_type,db_prefix,spec['src_uri'])
     #Check to which staging server the database need to be copied to
     (spec,staging_uri) = check_staging_server(spec,db_type,db_prefix,assembly)
     #setting compara url to default value for species databases. This value is only used by Compara healthchecks
-    if (compara_uri == None):
+    if compara_uri == None:
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    if compara_uri is None:
```

File: ensembl_prodinf/handover_tasks.py [hunk 2]
```diff
@@ -95,25 +100,25 @@ def get_tgt_uri(src_url,staging_uri):
     
 def check_db(uri):    
     """Check if source database exists"""
-    if(database_exists(uri) == False):
+    if database_exists(uri) == False:
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    if not database_exists(uri):
```

File: ensembl_prodinf/handover_tasks.py [hunk 3]
```diff
@@ -181,20 +207,20 @@ def process_checked_db(self, hc_job_id, spec):
     self.max_retries = None
     get_logger().info("HCs in progress, please see: " +cfg.hc_web_uri + str(hc_job_id))
     result = hc_client.retrieve_job(hc_job_id)
-    if (result['status'] == 'incomplete') or (result['status'] == 'running') or (result['status'] == 'submitted'):
+    if result['status'] == 'incomplete' or result['status'] == 'running' or result['status'] == 'submitted':
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    if result['status'] in ['incomplete', 'running', 'submitted']:
```

File: ensembl_prodinf/handover_tasks.py [hunk 4]
```diff
@@ -227,18 +263,18 @@ def process_copied_db(self, copy_job_id, spec):
     self.max_retries = None
     get_logger().info("Copying in progress, please see: " +cfg.copy_web_uri + str(copy_job_id))
     result = db_copy_client.retrieve_job(copy_job_id)
-    if (result['status'] == 'incomplete') or (result['status'] == 'running') or (result['status'] == 'submitted'):
+    if result['status'] == 'incomplete' or result['status'] == 'running' or result['status'] == 'submitted':
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]:
```suggestion
    if result['status'] in ['incomplete', 'running', 'submitted']:
```

Comment (marcoooo) [reply ensembl_prodinf/handover_tasks.py]: As I just made the second same change, we must consider having some kind of 'Constants' somewhere, may be as property in a dedicated class for storing job results. Where we could centralize the behavior 'whether or not the job is completed, failed etc...' As a start we can at least register this list at the beginning of the file then only once.

```
_completed_status = ['incomplete', 'running', 'submitted']
...
if result['status'] in _completed_status:
```

File: ensembl_prodinf/handover_tasks.py [hunk 5]
```diff
@@ -265,10 +301,10 @@ def process_db_metadata(self, metadata_job_id, spec):
     self.max_retries = None
     get_logger().info("Loading into metadata database, please see: "+cfg.meta_uri + "jobs/"+ str(metadata_job_id))
     result = metadata_client.retrieve_job(metadata_job_id)
-    if (result['status'] == 'incomplete') or (result['status'] == 'running') or (result['status'] == 'submitted'):
+    if result['status'] == 'incomplete' or result['status'] == 'running' or result['status'] == 'submitted':
```
Comment (marcoooo) [inline ensembl_prodinf/handover_tasks.py]: Same comment
```suggestion
    if result['status'] in ['incomplete', 'running', 'submitted']:
```

File: ensembl_prodinf/handover_tasks.py (hunk 4 above)
Comment (thomasmaurel) [reply ensembl_prodinf/handover_tasks.py]: Sure, that would make sense

**Review by marcoooo — APPROVED** [review]

**Pull request merged**